v2/pkg/engine/datasource/graphql_datasource/graphql_subscription_client.go

```diff
@@ -112,6 +112,9 @@ func NewGraphQLSubscriptionClient(httpClient, streamingClient *http.Client, engi
 // If connection protocol is SSE, a new connection is always created
 // If no connection exists, the client initiates a new one
 func (c *subscriptionClient) Subscribe(reqCtx *resolve.Context, options GraphQLSubscriptionOptions, updater resolve.SubscriptionUpdater) error {
+	if reqCtx.ExecutionOptions.SendHeartbeat {
```

Comment:
I'm not sure if this is the right approach. If the first subscription is not multipart while the second one is deduplicated, it could result in no heartbeat being sent. You would need to run the heartbeat continuously or enable the heartbeat before the duplication check.

Reply:
What else do you think makes sense to do? From what I could see in the `handleAddSubscription`, after ensuring that the triggerID is present, it will add a subscription to the map and then return, without calling `add.resolve.Trigger.Source.Start`. Should it instead always send the heartbeat, and then in `handleHeartbeat`, when looping over all of the `trig.subscriptions`, it would only send it to the ones who have `sub.sendHeartbeat` as `true`? 

My only concern is performance. That change would mean that every subscription will be sending a `heartbeat`, and then on the resolver side it's only sent if the subscription has opted in. That could conceivably mean significantly more events for the event handler to handle that users hadn't opted into. Do you think it's worth the potential performance issue, am I just exaggerating it?

Reply:
> What else do you think makes sense to do? From what I could see in the handleAddSubscription, after ensuring that the triggerID is present, it will add a subscription to the map and then return, without calling add.resolve.Trigger.Source.Start. Should it instead always send the heartbeat, and then in handleHeartbeat, when looping over all of the trig.subscriptions, it would only send it to the ones who have sub.sendHeartbeat as true?

For me, it sounds fine, especially when considering that heartbeats are only sent when no data has been recently sent on the trigger subscriptions.